c10/core/impl/alloc_cpu.cpp

```diff
@@ -41,6 +41,38 @@ void memset_junk(void* data, size_t num) {
   }
 }
 
+static inline bool is_thp_alloc_enabled() {
+  static bool value = [&](const char* pt) {
+    if (pt != NULL) {
+      return std::atoi(pt);
+    } else {
+      return 0;
+    }
+  }(std::getenv("THP_MEM_ALLOC_ENABLE"));
+  return value;
+}
+
+#ifdef __linux__
+inline size_t c10_compute_alignment(size_t nbytes) {
+  static int pagesize = sysconf(_SC_PAGESIZE);
+  // for kernels that don't provide page size, default it to 4K
+  const size_t thp_alignment = (gPagesize < 0 ? gPagesize : pagesize);
+  return (is_thp_alloc_enabled() ? thp_alignment : gAlignment);
+}
+
+inline bool is_thp_alloc(size_t nbytes) {
+  // enable thp (transparent huge pages) for larger buffers
+  return (is_thp_alloc_enabled() && (nbytes >= gAlloc_threshold_thp));
+}
+#else
+constexpr size_t c10_compute_alignment(size_t nbytes) {
+  return gAlignment;
+}
+
+constexpr bool is_thp_alloc(size_t nbytes) {
```

Comment:
Aren't this one needed to suppress compilation warnings on say MacOS?
```suggestion
constexpr bool is_thp_alloc(C10_UNUSED size_t nbytes) {
```